tf/tfprocess.py

```diff
@@ -128,191 +113,167 @@ def __init__(self, cfg):
-            gradient[0]) for accum, gradient in zip(gradient_accum, gradients)]
-        # gradients are num_batch_splits times higher due to accumulation by summing, so the norm will be too
-        max_grad_norm = self.cfg['training'].get(
-            'max_grad_norm', 10000.0) * self.cfg['training'].get('num_batch_splits', 1)
-        gradient_accum, self.grad_norm = tf.clip_by_global_norm(
-            gradient_accum, max_grad_norm)
-        self.train_op = opt_op.apply_gradients(
-            [(accum, gradient[1]) for accum, gradient in zip(gradient_accum, gradients)], global_step=self.global_step)
-
-        correct_policy_prediction = \
-            tf.equal(tf.argmax(self.y_conv, 1), tf.argmax(self.y_, 1))
-        correct_policy_prediction = tf.cast(correct_policy_prediction, tf.float32)
-        self.policy_accuracy = tf.reduce_mean(correct_policy_prediction)
-        correct_value_prediction = \
-            tf.equal(tf.argmax(self.z_conv, 1), tf.argmax(self.z_, 1))
-        correct_value_prediction = tf.cast(correct_value_prediction, tf.float32)
-        self.value_accuracy = tf.reduce_mean(correct_value_prediction)
+        self.lossMix = lambda policy, value: pol_loss_w * policy + val_loss_w * value
 
+        def accuracy(target, output):
+            output = tf.cast(output, tf.float32)
+            return tf.reduce_mean(tf.cast(tf.equal(tf.argmax(input=target, axis=1), tf.argmax(input=output, axis=1)), tf.float32))
+        self.accuracy_fn = accuracy
+       
         self.avg_policy_loss = []
         self.avg_value_loss = []
         self.avg_mse_loss = []
         self.avg_reg_term = []
         self.time_start = None
         self.last_steps = None
-
-        # Summary part
-        self.test_writer = tf.summary.FileWriter(
+        # Set adaptive learning rate during training
+        self.cfg['training']['lr_boundaries'].sort()
+        self.warmup_steps = self.cfg['training'].get('warmup_steps', 0)
+        self.lr = self.cfg['training']['lr_values'][0]
+        self.test_writer = tf.summary.create_file_writer(
             os.path.join(os.getcwd(), "leelalogs/{}-test".format(self.cfg['name'])))
-        self.train_writer = tf.summary.FileWriter(
+        self.train_writer = tf.summary.create_file_writer(
             os.path.join(os.getcwd(), "leelalogs/{}-train".format(self.cfg['name'])))
         if self.swa_enabled:
-            self.swa_writer = tf.summary.FileWriter(
+            self.swa_writer = tf.summary.create_file_writer(
                 os.path.join(os.getcwd(), "leelalogs/{}-swa-test".format(self.cfg['name'])))
-        self.histograms = [tf.summary.histogram(
-            weight.name, weight) for weight in self.weights]
-
-        self.init = tf.global_variables_initializer()
-        self.saver = tf.train.Saver()
-
-        self.session.run(self.init)
+        self.checkpoint = tf.train.Checkpoint(optimizer=self.orig_optimizer, model=self.model, global_step=self.global_step, swa_count=self.swa_count)
```

Comment:
This line fails when `swa: false` is set since it is set to None and isn't a trackable object.

Reply:
Works when `self.swa_count` definition above is taken out of the if clause.

Reply:
Done.